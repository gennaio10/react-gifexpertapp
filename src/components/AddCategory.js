import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleOnSubmit = (e) => {
    //evita que se recarga la pagina
    e.preventDefault();

    if (inputValue.trim().length > 1) {
      setCategories((categoriesE) => [inputValue, ...categoriesE]);
      setInputValue("");
    }
  };
  
  return (
    <form onSubmit={handleOnSubmit}>
      <h2>Add Category</h2>
      <input id="icategoria" type="text" value={inputValue} onChange={handleChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
