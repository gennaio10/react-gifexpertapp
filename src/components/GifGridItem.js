import React from "react";
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
  return (
    <div id="dimg" className="card animate__animated animate__bounce">
      <img id="isrc" src={url} alt={title} />
      <p id="ptitle">{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};



