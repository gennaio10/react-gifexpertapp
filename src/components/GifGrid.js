import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data:imagenes, loadind } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loadind && <p>Loading...</p>}
      <div className="card-grid">
        {imagenes.map((imagenConUrl) => (
          <GifGridItem key={imagenConUrl.id} {...imagenConUrl} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
