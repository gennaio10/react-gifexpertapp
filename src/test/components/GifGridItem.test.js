import React from "react";
import "@testing-library/jest-dom";
import { GifGridItem} from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Pruebas de <GifGridItem />)", () => {
  const title = "Un titulo";
  const url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test(" debe mostrar el componenete correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test(" debe de tener un parrafo con el titulo enviado", () => {
    const textoParrafo = wrapper.find('#ptitle');
    expect(textoParrafo.text()).toBe(title);
  });

  test(" debe de tener un img con la url y alt enviados", () => {
    const urlImg = wrapper.find('#isrc');
    expect(urlImg.props().src).toBe(url);
    expect(urlImg.props().alt).toBe(title);
  });

  test(" debe de tener un img con la url y alt enviados", () => {
    const urlImg = wrapper.find('#dimg');
    const claseBuscada = 'animate__bounce';
    expect(urlImg.props().className.includes(claseBuscada)).toBe(true);
  });

});
