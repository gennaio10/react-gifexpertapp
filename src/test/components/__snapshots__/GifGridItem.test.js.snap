// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem />)  debe mostrar el componenete correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
  id="dimg"
>
  <img
    alt="Un titulo"
    id="isrc"
    src="https://localhost/algo.jpg"
  />
  <p
    id="ptitle"
  >
    Un titulo
  </p>
</div>
`;
